go/test/endtoend/tabletmanager/tablet_test.go

```diff
@@ -43,14 +43,16 @@ func TestEnsureDB(t *testing.T) {
 	err = clusterInstance.StartVttablet(tablet, "NOT_SERVING", false, cell, "dbtest", hostname, "0")
 	require.NoError(t, err)
 
+	assert.Equal(t, "NOT_SERVING", tablet.VttabletProcess.GetTabletStatus())
 	// Make it the primary.
 	err = clusterInstance.VtctlclientProcess.ExecuteCommand("TabletExternallyReparented", tablet.Alias)
-	require.EqualError(t, err, "exit status 1")
+	require.NoError(t, err, "No error expected.")
 
-	// It is still NOT_SERVING because the db is read-only.
-	assert.Equal(t, "NOT_SERVING", tablet.VttabletProcess.GetTabletStatus())
+	// Once promoted to primary during vschema/engine.go we change tablet to ReadWrite, hence promoting
+	// tablet to SERVING status.
+	assert.Equal(t, "SERVING", tablet.VttabletProcess.GetTabletStatus())
 	status := tablet.VttabletProcess.GetStatusDetails()
-	assert.Contains(t, status, "read-only")
+	assert.Contains(t, status, "healthy")
```

Comment:
FWIW, I think this is a JSON document so would be better to look at the given key.